shift/src/main/java/org/coursera/android/shift/ShiftPref.java

```diff
@@ -13,14 +13,38 @@
 
 package org.coursera.android.shift;
 
-interface ShiftPref<T> {
-    T getValue();
+abstract class ShiftPref <T> {
+    protected final ShiftPersistenceManager PERSISTENCE;
+    protected final String KEY;
+    protected final T DEFAULT_VALUE;
+    protected final Class<T> CLASS;
 
-    void setValue(T value);
+    public ShiftPref(ShiftPersistenceManager persistenceManager,
+                     String key,
+                     T defaultValue,
+                     Class<T> myClass) {
+        PERSISTENCE = persistenceManager;
+        KEY = key;
+        DEFAULT_VALUE = defaultValue;
+        CLASS = myClass;
+        if (!isValueSet()) {
```

Comment:
@BenSchwab I added this because otherwise every time you quit the application and relaunch, it will use the default values instead of the ones you have saved.